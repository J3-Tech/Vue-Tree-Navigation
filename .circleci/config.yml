version: 2
jobs:
  build:
    docker:
      - image: circleci/node:8

    steps:
      - checkout

      - restore_cache:
          keys:
          - v1-dependencies-{{ checksum "package.json" }}
          - v1-dependencies-

      - run: yarn install

      - save_cache:
          paths:
            - node_modules
          key: v1-dependencies-{{ checksum "package.json" }}

      - run: yarn prettier-list
      - run: yarn lint
      - run: yarn unit-ci
