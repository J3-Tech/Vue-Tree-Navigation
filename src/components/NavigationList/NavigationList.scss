.NavigationList--closed {
  ul,
  li {
    display: none;
  }
}
