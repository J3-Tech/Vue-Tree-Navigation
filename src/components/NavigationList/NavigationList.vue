<template>
  <ul class="NavigationList" :class="classes">
    <div @click="toggle">
      <NavigationToggle :isClosed="closed" />
      <NavigationItem :item="parentItem" />
    </div>

    <!-- child items goes here -->
    <slot></slot>
  </ul>
</template>

<script>
import NavigationToggle from '../NavigationToggle/NavigationToggle.vue';
import NavigationItem from '../NavigationItem/NavigationItem.vue';

export default {
  data() {
    return {
      closed: this.level > this.defaultOpenLevel,
    };
  },

  props: {
    level: {
      type: Number,
      required: true,
    },
    defaultOpenLevel: {
      type: Number,
      required: true,
    },
    parentItem: {
      type: Object,
      required: true,
    },
  },

  computed: {
    classes() {
      return {
        'NavigationList--closed': this.closed,
        [`NavigationList--level-${this.level}`]: true  
      };
    },
  },

  watch: {
    defaultOpenLevel: function() {
      this.closed = this.level > this.defaultOpenLevel;
    },
  },

  methods: {
    toggle() {
      this.closed = !this.closed;
    },
  },

  components: {
    NavigationItem,
    NavigationToggle,
  },
};
</script>

<style lang="scss">
@import './NavigationList.scss';
</style>
