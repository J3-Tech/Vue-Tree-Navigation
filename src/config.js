export const PATH_TYPE_NONE = 'none';
export const PATH_TYPE_ELEMENT = 'element';
export const PATH_TYPE_ROUTE = 'route';
export const PATH_TYPE_EXTERNAL = 'external';
