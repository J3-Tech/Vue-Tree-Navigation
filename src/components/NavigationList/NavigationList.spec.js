import { shallow } from 'vue-test-utils';

import { PATH_TYPE_NONE } from '../../config';
import NavigationList from './NavigationList';

const parentItem = {
  name: 'Contact',
  meta: {
    pathType: PATH_TYPE_NONE,
  },
};

describe('NavigationList ', () => {
  it('is Vue instance', () => {
    const wrapper = shallow(NavigationList, {
      propsData: {
        level: 2,
        defaultOpenLevel: 3,
        parentItem,
      },
    });

    expect(wrapper.isVueInstance()).toBe(true);
  });

  context('with level greater than default open level', () => {
    it('assigns closed class to a list`', () => {
      const wrapper = shallow(NavigationList, {
        propsData: {
          level: 3,
          defaultOpenLevel: 2,
          parentItem,
        },
      });

      expect(wrapper.classes()).toContain('NavigationList--closed');
    });
  });

  context('with level equal to default open level', () => {
    it('does not assign closed class to a list', () => {
      const wrapper = shallow(NavigationList, {
        propsData: {
          level: 3,
          defaultOpenLevel: 3,
          parentItem,
        },
      });

      expect(wrapper.classes()).not.toContain('NavigationList--closed');
    });
  });

  context('with level less than default open level', () => {
    it('does not assign closed class to a list', () => {
      const wrapper = shallow(NavigationList, {
        propsData: {
          level: 2,
          defaultOpenLevel: 3,
          parentItem,
        },
      });

      expect(wrapper.classes()).not.toContain('NavigationList--closed');
    });
  });
});
