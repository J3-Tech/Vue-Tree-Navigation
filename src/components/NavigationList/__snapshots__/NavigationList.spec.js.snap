// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationList  renders 1`] = `
<ul class="NavigationList">
  <div><span class="navigation-toggle"><span class="navigation-toggle-icon"></span></span> <span class="NavigationItem"><span>Contact</span>
    <!---->
    <!---->
    </span>
  </div>
</ul>
`;
